launcher/launcher.go

```diff
@@ -445,7 +445,19 @@ func (c *Config) getTracesEndpoint() (string, bool) {
 		c.TracesExporterProtocol = c.ExporterProtocol
 	}
 
-	// use traces specific port, failling back to generic version if not set
+	// helper function - if using grpc and prepending with http, drop the http scheme
+	if c.TracesExporterProtocol == ProtocolGRPC {
+		if strings.HasPrefix(c.TracesExporterEndpoint, "https://") {
+			c.TracesExporterEndpoint = strings.TrimPrefix(c.TracesExporterEndpoint, "https://")
+			newEndpoint := strings.Join([]string{c.TracesExporterEndpoint, ":443"}, "")
```

Comment:
🤔 I think this Join may result in multiple ports appearing on an endpoint string if the incoming config has a port declared:

- in: `https://our.collector.with.an.actual.cert:1337`
- out: `our.super.secure.proxy:1337:443`

Reply:
Hm, yeah that's a great point that I hadn't considered. I wonder if I adjust this to split host and port and check for existence of a port before appending. Will look into that now.

Reply:
[url.Parse](https://pkg.go.dev/net/url#Parse) from standard lib to [chop an endpoint up into its named pieces](https://golangbyexample.com/parse-url-golang/)?

Reply:
I've updated the logic to cover that use case, and rearranged things a bit.

Reply:
I might be able to clean up those functions a bit , as I just created 2 new functions dedicated to parsing endpoints and left the previous function alone to still do its thing 🤔